includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2912,6 +2944,30 @@ static function ( $selector_language ) {
 		}
 	}
 
+	/**
+	 * Given list of class names, create a regular expression pattern to match them in a selector.
+	 *
+	 * @since 1.4
+	 *
+	 * @param string[] $class_names Class names.
+	 * @return string Pattern,
+	 */
+	private static function get_class_name_selector_pattern( $class_names ) {
+		$pattern  = '/\.(';
```

Comment:
Although it is pretty irrelevant at the scale here, building a pattern in this way is very wasteful, with multiple assignments and memory concatenations.

I think the following structure would make it both more efficient and more readable:
```php
$class_pattern = implode(
	'|',
	array_map(
		static function ( $class_name ) {
			return preg_quote( $class_name, '/' );
		},
		$class_names
	)
);
return "/\.({$class_pattern})(?=$|[^a-zA-Z0-9_-])/";

Reply:
Done in 8cc8e81e. If you like, you can also add an additional CLA consent 😄